lib/samson/process_utils.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Samson
+  module ProcessUtils
+    ATTRIBUTES = ['pid', 'ppid', 'gid', 'pcpu', 'user', 'start', 'args'].freeze
+    WHITELISTS = ['puma'].freeze
+
+    class << self
+      def ps_list
+        pipe = IO.popen("ps -eo #{ATTRIBUTES.join(',')} | grep -v #{WHITELISTS.join(',')}")
+        pipe.readlines[1..-1].map do |line|
+          Hash[ATTRIBUTES.zip line.lstrip.split(/\s+/, 7)]
+        end
+      end
+
+      def report_to_statsd
+        ps_list.each do |process|
+          timeout = Time.parse(process.fetch('start')) < Time.now - Rails.application.config.samson.deploy_timeout
+          tags = ATTRIBUTES.map { |attr| "#{attr}:#{process.fetch(attr)}" }
```

Comment:
pass them in as an array (no hash coversion) and then `ATTRIBUTES.zip(process).map { |k, v| "#{k}:#{v}" }`

Reply:
It's hard to access attributes without hash conversion, I hope HASH formation shouldn't create much difference.